### PR TITLE
feat(tool): Add ability to view all open PRs for repos other than freeCodeCamp in the Dashboard app

### PR DESCRIPTION
**Pull request by RandellDawson**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Related to  #40407 (closed in favor of this PR)

This PR creates two new views on the Dashboard App (`Boilerplate PRs` and `Other Repo's PRs` that have at least one open PR. This will make it easier to quickly view the open PRs in each repo instead of having to go into each one to see them. In addition to adding these two new views, I stripped Probot from the app, renamed `/tools/dashboard` to `/tools/contributor` since the dashboard app (now located in `/tools/contributor/dashboard-app`) was just one part of this particular folder. Removing Probot greatly simplifies the ability to contribute to the app now.  Plus, with my restructuring, we can now proceed to make use of GitHub for various moderator tasks (i.e. labeling of PRs). I plan to clean up or remove a few of the one-off scripts that make no longer be relevant in a future PR.  Currently, we are not using any of the one-off scripts.

image

### CODE REVIEW

**Comment by scissorsneedfoodtoo** [comment]
@RandellDawson, the logo isn't in the CDN now, and I wasn't able to import the one that we use in the main client. Instead I just copied the `freeCodeCampLogo.js` file into `tools/contributor/dashboard-app/client/src/assets/freeCodeCampLogo.js` and used it as a component in `App.js`. 

There may be a better way to do this with Learna, a symlink, or moving the SVG file to the CDN. But I figure it works for now.

File: tools/contributor/docs/README.md
Comment (nhcarrigan) [inline tools/contributor/docs/README.md:24]:
```suggestion
node dashboard-app/server/tools/update-db.js
```
File name is different here 🙂

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
There's no diff so I can't comment on the line, but the base [README.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/be98f314f3968b436762c831bc761e607ef31225/tools/contributor/README.md) still links to the `gitter` contributors room - should that be updated?

Also, still using the green favicon - do we want to replace that in this PR or handle after this is merged?

Aside from these questions plus small nitpick below, this looks great - played around with it locally a bit, and everything seems to work well.

**Comment by RandellDawson** [comment]
> There's no diff so I can't comment on the line, but the base README.md still links to the gitter contributors room - should that be updated?

Sure.  Push a commit.

> Also, still using the green favicon - do we want to replace that in this PR or handle after this is merged?

Since I have not yet resolved the linting error yet, feel free to push a commit replacing the favicon.

**Comment by nhcarrigan** [comment]
I pushed the commits for my two small fixes.

Looking at the `lint` script (and I'm not sure why moving the files would have broken this) it appears that the CI is unable to load the `react-app` config for ESLint. I'm thinking it might be an issue where that's not a dependency in our main `package.json` and the sub-packages don't get an install script called so it is throwing an error.

A potential fix would be to replace the config in the tool's `package.json` with the config from our root settings? Loading these settings into the `tools/contributor/package.json` fixed the error for me locally:

```json
  "eslintConfig": {
    "extends": ["../../../../.eslintrc-base.json", "prettier", "prettier/react"]
  },
```

**Comment by nhcarrigan** [comment]
image

One quick question - are the non-visited links intended to be the same colour as the regular text?

**Comment by RandellDawson** [comment]
> One quick question - are the non-visited links intended to be the same colour as the regular text?

No, it was originally supposed to be green (like the green showing on the current live version).  We can adjust it to be something else in a separate PR later.  It looks and works much better than  it used to.

Thanks for your help debugging the linting issue I was experiencing.  It is always something simple.

**Comment by nhcarrigan** [comment]
Always glad to help!

**Comment by raisedadead** [comment]
@RandellDawson quick question? If we hypothetically used NGINX to serve the contribute site, can we re-wire this dashboard to be available at `https://contribute.freecodecamp.org/tools/` ?

What do you think?

**Comment by RandellDawson** [comment]
@nhcarrigan Are you ok with the changes now?  @moT01 @scissorsneedfoodtoo Can I get a second review on this PR?  I would like to start using the new dashboard features if possible.

Thanks.

**Comment by RohiniMallem** [comment]
Hi Freecodecamp/Freecodecamp,

I'm interested PRs for repos..

**Review by naomi-lgbt — DISMISSED** [review]
LGTM~! 🥳 Explored the new functionality and am very happy with these changes.

Thank you for your pull request! We look forward to additional contributions!

**Comment by nhcarrigan** [comment]
Hey there @RohiniMallem~! I recommend reviewing our [contributing guidelines](https://contribute.freecodecamp.org) to understand the process for working on our codebase. 🙂

**Review by scissorsneedfoodtoo — DISMISSED** [review]
Tested locally and everything LGTM :+1: :+1: 

Looking forward to using this revamped dashboard!

Edit: Might have spoken a bit too soon. There are not tests at the moment, so this isn't a big deal. Just writing this out for visibility in the future.

Seems like running `npm run test` from `.../tools/contributor` throws some errors complaining about the version of `es-lint` and `babel-eslint`.

Tried fixing this locally, but following the instructions to remove `babel-eslint` from the root `package.json` file breaks linting throughout the project.

Adding `SKIP_PREFLIGHT_CHECK=true` to the `test` script in `.../tools/contributor/dashboard-app/client/package.json` prevents the error messages.

I'll leave that as a suggestion, but it's just a minor detail that can come later.

File: tools/contributor/dashboard-app/client/package.json
Comment (scissorsneedfoodtoo) [inline tools/contributor/dashboard-app/client/package.json:15]:
```suggestion
    "test": "SKIP_PREFLIGHT_CHECK=true react-scripts test",
```

**Review by scissorsneedfoodtoo — APPROVED** [review]

**Comment by scissorsneedfoodtoo** [comment]
Thanks for committing that, @RandellDawson. Will merge this now.

**Pull request merged**
